Add persistent gluster and restructure persistent storage

screeley44 opened the pull request:
Trying to add some details about Persistent Storage for Gluster - and rather than have a .adoc file side by side with the persistent_storage_nfs.adoc, thought it made more sense to subdir this topic, assuming we might have other volume type provider examples in the near future (ceph, cinder, etc...).  Not sure this is the correct approach, let me know what you think.

adellape commented on admin_guide/persistent_storage/persistent_storage_nfs.adoc:
```diff
@@ -78,7 +78,7 @@ servers and paths and the `*PersistentVolume*` API.
 can be its own export. Each export is one persistent volume. Kubernetes enforces
 unique names for persistent volumes, but the uniqueness of the NFS volume's
-server and path is up to the administrator.
+server and path is up to the administrator.  test
```
s/test//

:)

adellape commented on admin_guide/persistent_storage/persistent_storage_glusterfs.adoc:
```diff
@@ -0,0 +1,182 @@
+toc::[]
+
+== Overview
+OpenShift can utilize persistent storage using Distributed File Systems (DFS) like GlusterFS.  Some familiarity with Kuberenetes/Docker is assumed.  Also assumed that there is access to an existing GlusterFS cluster and volume and that the glusterfs-client has been installed on all OpenShift nodes in the cluster.
```
s/Kuberenetes/Kubernetes/

adellape commented on admin_guide/persistent_storage/persistent_storage_glusterfs.adoc:
```diff
@@ -0,0 +1,182 @@
+== Overview
+OpenShift can utilize persistent storage using Distributed File Systems (DFS) like GlusterFS.  Some familiarity with Kuberenetes/Docker is assumed.  Also assumed that there is access to an existing GlusterFS cluster and volume and that the glusterfs-client has been installed on all OpenShift nodes in the cluster.
+
+The Kubernetes link:../dev_guide/persistent_volumes.html[persistent volume]
```
Need to go up one more level.

adellape commented on admin_guide/persistent_storage/persistent_storage_glusterfs.adoc:
```diff
@@ -0,0 +1,182 @@
+
+[NOTE]
+====
+`Persistent Volumes` are not bound to a single project/namespace, they can be shared across the OpenShift environment, where as, `Persistent Volume Claims` are project/namespace specific and can be requested by users
```
s/users/users./

adellape commented on admin_guide/persistent_storage/persistent_storage_glusterfs.adoc:
```diff
@@ -0,0 +1,182 @@
+.Persistent Volume Endpoints Definition
+====
+
+For this example, you will have to define the GlusterFS Cluster as “EndPoints” within Kubernets/OpenShift platform using a file similar to [endpoints configuration file](gluster-endpoints.json).  You must define the IP/Hostname of your gluster servers and the port that you want to use.  The port value can be any numeric value within the accepted range of ports.
```
s/Kubernets/Kubernetes/

adellape replied:
Were you planning on including a link to an actual `gluster-endpoints.json` file, or is this basically just referring to the example JSON that now follows?

adellape commented on _build_cfg.yml:
```diff
@@ -167,8 +167,13 @@ Topics:
+      - Name: Persistent Storage Using NFS
+        File: persistent_storage_nfs
+      - Name: Persistent Storage Using GlusterFS
+        File: persistent_storage_glusterfs
```
Since the `Topics.Name` fields here can show a different title in the left nav than what appears in the actual `.adoc` files' title heading, how about for the left nav just have "Using NFS" and "Using GlusterFS", respectively, since it's a bit redundant w/ the "Persistent Storage" directory heading already right above it. And the title headings in the `adoc` files themselves could stay as-is (e.g., "Persistent Storage Using GlusterFS", etc).

adellape commented:
@screeley44 Thanks for this! Some comments inline, mostly about links, a couple minor typos. Also, for the NFS topic, the links need to be updated since the topic was moved to a different directory; currently any xrefs are broken.

After the above are taken care of, I can merge this and then do a pass on the topic for some style/formatting edits in a follow-up PR, and I'll ping you for review. After that merges, I'll then make sure the topic is tagged for inclusion in OSE in the appropriate release.

screeley44 commented:
@adellape  - thanks for the review, I think I addressed all of your comments, I didn't plan on having a link to any of the actual json files (endpoints, pods, etc...) since they can be copied easy enough from the doc and noticed NFS didn't have those either that I could tell.  I think if my other storage examples get merged into the origin examples, we can update this with a PR to point to the more detailed examples??

adellape commented:
@screeley44 This the origin PR to be keeping an eye on? https://github.com/openshift/origin/pull/4408

screeley44 commented:
Hi Alex,
Yeah, that is the origin PR that has some glusterfs examples and there is still the origin-docs pr that is not merged yet...https://github.com/openshift/openshift-docs/pull/924

thanks,
Scott

----- Original Message -----
From: "Alex Dellapenta" <notifications@github.com>
To: "openshift/openshift-docs" <openshift-docs@noreply.github.com>
Cc: "Scott Creeley" <screeley@redhat.com>
Sent: Thursday, September 17, 2015 7:30:54 PM
Subject: Re: [openshift-docs] Add persistent gluster and restructure persistent storage (#924)

@screeley44 This the origin PR to be keeping an eye on? https://github.com/openshift/origin/pull/4408

---
Reply to this email directly or view it on GitHub:
https://github.com/openshift/openshift-docs/pull/924#issuecomment-141276417

adellape commented:
@screeley44 Sorry, maybe I misunderstood earlier. I see https://github.com/openshift/origin/pull/4408 is still unmerged, but does that block my merging of this openshift-docs PR? If not, I'll get this merged since it introduces the restructured storage topics which https://github.com/openshift/openshift-docs/pull/930 also builds on. Then I'll follow up with any edits in a separate PR.

One other thing: could you convert the JSON to YAML? Trying to sync up on YAML for examples going forward.

screeley44 commented:
@adellape - no, this PR does not block the openshift/origin#4408  .  I will work on converting the 4408 json to yaml as well

adellape commented:
Merged this, and going to follow up now with fixes to links elsewhere in the docs that were pointing to the old location of the NFS topic.
